hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -660,9 +660,16 @@ func reconcileUserDataSecret(userDataSecret *corev1.Secret, nodePool *hyperv1.No
 	if err != nil {
 		return fmt.Errorf("failed to marshal ignition config: %w", err)
 	}
-	userDataSecret.Data = map[string][]byte{
-		"disableTemplating": []byte(base64.StdEncoding.EncodeToString([]byte("true"))),
-		"value":             userDataValue,
+	if nodePool.Spec.Platform.Type == hyperv1.AgentPlatform {
```

Comment:
This is not meant to be used as an API. It's a an impl detail and we won't introduce skew per platform, this is a design invariant.